### ISSUE TITLE
Doc(Transfer learning and fine-tuning) is quite different from real executive result.

### ISSUE THREAD

**Issue by lida2003** [tensorflow/tensorflow#66696]
### Issue type

Support

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

binary

### TensorFlow version

2.15.0+nv24.03 GPU version, check this link: https://forums.developer.nvidia.com/t/multiple-executive-warnings-after-switching-tensorflow-from-2-16-1-cpu-to-v60dp-tensorflow-2-15-0-nv24-03-gpu-version/291208

### Custom code

No

### OS platform and distribution

Jetson Orin Nano ubuntu 22.04 Jammy

### Python version

3.10.12

### CUDA/cuDNN version

CUDA12.2.140/cuDNN8.9.4.25

### GPU model and memory

sm90 8GB

### Current behavior?

The executive result (trend of curve and abosolute value) is quite different from document.

- My result: [transfer_learning.zip](https://github.com/tensorflow/tensorflow/files/15162333/transfer_learning.zip)

![图片](https://github.com/tensorflow/tensorflow/assets/1296171/708f72e2-30f4-4033-9e2b-d064886db4ff)

- Document (https://github.com/tensorflow/docs/blob/master/site/en/tutorials/images/transfer_learning.ipynb) says:

![图片](https://github.com/tensorflow/tensorflow/assets/1296171/99489450-da99-4529-a599-516682bba858)

![图片](https://github.com/tensorflow/tensorflow/assets/1296171/7834ab25-25ae-4443-8775-869905cf7d01)

### Standalone code to reproduce the issue

```shell
100% reproducable
```

### Relevant log output

Also tried Colab, which is consistent with documentation:

![图片](https://github.com/tensorflow/tensorflow/assets/1296171/5b474c91-805c-4d00-8158-294981a8fc0f)

**Comment by tilakrayal** [comment]
@lida2003,
Could you please confirm whether the difference in Accuracy & Loss has happened in GPU/CPU with both tensorflow v2.15, v2.16? Also I will also try to debug more on this issue and provide the resolution. Thank you!

**Comment by lida2003** [comment]
> @lida2003,
> Could you please confirm whether the difference in Accuracy & Loss has happened in GPU/CPU with both tensorflow v2.15, v2.16? 


- NVIDIA 2.15.0+nv24.03 GPU version faild 100%
- Colab is consistent with tensorflow document (But I don't know the version)

PS: Jetson Orin Nano 8GB, CPU&GPU shared memory


> Also I will also try to debug more on this issue and provide the resolution. Thank you!

There are also copule of other things might be a clue for you, link: Here is a link on NVIDIA forum: https://forums.developer.nvidia.com/t/tensorflow-v2-16-1-gpu-version-local-build-on-jetson-orin-nano-failed/291195/7?u=daniel2008_12

And the resource issue, we have confirmed with NVIDIA: https://forums.developer.nvidia.com/t/multiple-executive-warnings-after-switching-tensorflow-from-2-16-1-cpu-to-v60dp-tensorflow-2-15-0-nv24-03-gpu-version/291208
As you can see there is not resource warning in the above demo, but might happen, which we don't know yet.

**Comment by tilakrayal** [comment]
@lida2003,
As you mentioned, I tried to execute the transfer learning code on the colab with the Tensorflow v2.16, v2.15 and observed that the loss and Accuracy is the same as the document. I suspect there might be an issue with the GPU. I will re-check and deep dive on the same. 
![image](https://github.com/tensorflow/tensorflow/assets/81610181/e1c45818-1619-4ddf-9d52-f4639bdac90d)


Thank you!

**Comment by lida2003** [comment]
@tilakrayal 

Please check [Inconsistency of NVIDIA 2.15.0+nv24.03 v.s. Colab v.s. Tensorflow Documentation](https://forums.developer.nvidia.com/t/inconsistency-of-nvidia-2-15-0-nv24-03-v-s-colab-v-s-tensorflow-documentation/291613)

- JetPack 6.0 DP and tf2.15.0+nv24.04  ==> BAD
- JetPack v6.0 + tensorflow2.15.0+nv24.05  ==>OK

**Comment by tilakrayal** [comment]
@lida2003,
Could you please let me know if you are facing the same issue on TensorFlow v2.16 where keras3.0 is by default included. eras 3 is a full rewrite of Keras that enables you to run your Keras workflows on top of either JAX, TensorFlow, or PyTorch, and that unlocks brand new large-scale model training and deployment capabilities.

Thank you!

**Comment by lida2003** [comment]
> Could you please let me know if you are facing the same issue on TensorFlow v2.16 where keras3.0 is by default included.

Sorry, I have met difficulties on build from source code (I have asked about some build steps, but I built without any luck)


- https://forums.developer.nvidia.com/t/how-to-compile-tensorflow-for-jetson-orin-nano/291321
- https://forums.developer.nvidia.com/t/tensorflow-v2-16-1-gpu-version-local-build-on-jetson-orin-nano-failed/291195


so I have used NVIDIA binary. 

- [2.15 is their latest build for v60](https://developer.download.nvidia.com/compute/redist/jp/v60/tensorflow/)

![图片](https://github.com/tensorflow/tensorflow/assets/1296171/f6f6388c-5a22-4822-9b94-2e2cab26cb74)

- [2.15 is their latest build for v60dp](https://developer.download.nvidia.com/compute/redist/jp/v60dp/tensorflow/)

![图片](https://github.com/tensorflow/tensorflow/assets/1296171/d08f9d86-fe5e-4dfa-858c-0efd9facc07d)


But I think it might be some issue with NVIDIA V60DP version, check this for details: [Inconsistency of NVIDIA 2.15.0+nv24.03 v.s. Colab v.s. Tensorflow Documentation](https://forums.developer.nvidia.com/t/inconsistency-of-nvidia-2-15-0-nv24-03-v-s-colab-v-s-tensorflow-documentation/291613)

**Comment by tilakrayal** [comment]
@lida2003,
Is this happening with the latest tensorflow v2.17 and the keras-nightly as well. Please try to test and let us know if you are facing the same issue. Thank you!

**Comment by lida2003** [comment]
> Is this happening with the latest tensorflow v2.17 and the keras-nightly as well.

I will try if there is a 2.17 release, but right now. It's NOT supported from Jetson binary release.

![图片](https://github.com/user-attachments/assets/e123d348-be9a-4795-8361-46fbbe75736c)

**Comment by tilakrayal** [comment]
@lida2003,
https://developer.download.nvidia.com/compute/redist/jp/v60/tensorflow/  Tensorflow is not handling this files. COuld you please check with the Nvidia team on the 2.17 whl. Thank you!

**Comment by lida2003** [comment]
OK, I have made a post for this. But I'm NOT sure when they will provide a binary for me to test.

https://forums.developer.nvidia.com/t/is-there-tensorflow-2-17-supported-by-jetpack-6-0/303264

**Comment by lida2003** [comment]
Thanks for remind.

I checked navidia repo https://developer.download.nvidia.com/compute/redist/jp/v60/tensorflow/  
As forum support, I didn't see any binary 2.17 release yet.

**Comment by lida2003** [comment]
> Are you satisfied with the resolution of your issue?
> <a href="https://docs.google.com/forms/d/e/1FAIpQLSfaP12TRhd9xSxjXZjcZFNXPGk4kc1-qMdv3gc6bEP90vY1ew/viewform?entry.85265664=Yes&entry.2137816233=https://github.com/tensorflow/tensorflow/issues/66696">Yes</a>
> <a href="https://docs.google.com/forms/d/e/1FAIpQLSfaP12TRhd9xSxjXZjcZFNXPGk4kc1-qMdv3gc6bEP90vY1ew/viewform?entry.85265664=No&entry.2137816233=https://github.com/tensorflow/tensorflow/issues/66696">No</a>
> 

No response from  Nvidia binary release, right now.
